### ISSUE TITLE
Move crypto package out of core

### ISSUE THREAD

**Issue by mkalinin** [ethereum/ethereumj#1130]
It would be pretty useful for developers to have `crypto` package separated from core and maintained as an independent module with its own artifact.

**Comment by RichardH92** [comment]
@mkalinin Would this involve making a separate gradle poject for just `crypto` but leaving in the same repo?

**Comment by mkalinin** [comment]
Yep. `crypto` should be a separate module with its own dependencies and tests like `ethereumj-core` and having its own publishing process. Then it will make sense to have core module dependent on `crypto`. It also make sense if `crypto` module will have a facade describing its main entry points, so, it will be possible to implement custom module which will serve as a crypto features provider.
